Mathlib/MeasureTheory/Function/LocallyIntegrable.lean

```diff
@@ -389,6 +389,58 @@ theorem integrable_iff_integrableAtFilter_cocompact :
   rewrite [← integrableOn_univ, ← compl_union_self s, integrableOn_union]
   exact ⟨(hloc.integrableOn_isCompact htc).mono ht le_rfl, hs⟩
 
+theorem integrable_iff_integrableAtFilter_atBot_atTop [LinearOrder X] [CompactIccSpace X] :
+    Integrable f μ ↔
+    (IntegrableAtFilter f atBot μ ∧ IntegrableAtFilter f atTop μ) ∧ LocallyIntegrable f μ := by
+  constructor
+  · exact fun hf ↦ ⟨⟨hf.integrableAtFilter _, hf.integrableAtFilter _⟩, hf.locallyIntegrable⟩
+  · refine fun h ↦ integrable_iff_integrableAtFilter_cocompact.mpr ⟨?_, h.2⟩
+    exact (IntegrableAtFilter.sup_iff.mpr h.1).filter_mono cocompact_le_atBot_atTop
+
+theorem integrable_iff_integrableAtFilter_atBot [LinearOrder X] [OrderTop X] [CompactIccSpace X] :
+    Integrable f μ ↔ IntegrableAtFilter f atBot μ ∧ LocallyIntegrable f μ := by
+  constructor
+  · exact fun hf ↦ ⟨hf.integrableAtFilter _, hf.locallyIntegrable⟩
+  · refine fun h ↦ integrable_iff_integrableAtFilter_cocompact.mpr ⟨?_, h.2⟩
+    exact h.1.filter_mono cocompact_le_atBot
+
+theorem integrable_iff_integrableAtFilter_atTop [LinearOrder X] [OrderBot X] [CompactIccSpace X] :
+    Integrable f μ ↔ IntegrableAtFilter f atTop μ ∧ LocallyIntegrable f μ :=
+  integrable_iff_integrableAtFilter_atBot (X := Xᵒᵈ)
+
+variable {a : X}
+
+theorem integrableOn_Iic_iff_integrableAtFilter_atBot [LinearOrder X] [CompactIccSpace X] :
+    IntegrableOn f (Iic a) μ ↔ IntegrableAtFilter f atBot μ ∧ LocallyIntegrableOn f (Iic a) μ := by
+  refine ⟨fun h ↦ ⟨⟨Iic a, Iic_mem_atBot a, h⟩, h.locallyIntegrableOn⟩, fun ⟨⟨s, hsl, hs⟩, h⟩ ↦ ?_⟩
+  haveI : Nonempty X := Nonempty.intro a
+  obtain ⟨a', ha'⟩ := mem_atBot_sets.mp hsl
+  refine (integrableOn_union.mpr ⟨hs.mono ha' le_rfl, ?_⟩).mono Iic_subset_Iic_union_Icc le_rfl
+  exact h.integrableOn_compact_subset Icc_subset_Iic_self isCompact_Icc
+
+theorem integrableOn_Ici_iff_integrableAtFilter_atTop [LinearOrder X] [CompactIccSpace X] :
+    IntegrableOn f (Ici a) μ ↔ IntegrableAtFilter f atTop μ ∧ LocallyIntegrableOn f (Ici a) μ :=
+  integrableOn_Iic_iff_integrableAtFilter_atBot (X := Xᵒᵈ)
+
+theorem integrableOn_Iio_iff_integrableAtFilter_atBot_nhdsWithin
+    [LinearOrder X] [CompactIccSpace X] [NoMinOrder X] [OrderTopology X] :
```

Comment:
Do you really need the `NoMinOrder` assumption here? I would guess the theorem is also true otherwise.

Reply:
If it’s true without the assumption, we should prove it in that generality, whether or not the corner case is mathematically interesting – there is benefit in having a shorter and simpler set of hypotheses.

Reply:
I guess what I'm saying is that I'm not convinced I can remove the assumption without adding `[MeasurableSingletonClass X]` and `μ {a} ≠ ∞` assumptions as per `integrableOn_singleton_iff`. 

I think, due to `IntegrableAtFilter f atBot μ ↔ IntegrableOn {⊥} μ` in the bottom element case, that the true theorems would be:
- the theorem as stated
- a different theorem `IntegrableOn f (Iio a) μ ↔ IntegrableAtFilter f atBot μ ∧ LocallyIntegrableOn f (Iio a) μ` which holds when `[MeasurableSingletonClass X]` and `μ {a} ≠ ∞`

At least, I haven't been successful in proving the `IntegrableOn f (Iio a) μ → IntegrableAtFilter f atBot μ` case without `[NoMinOrder X]`, although I haven't disproven it either.

Reply:
Ah, you're right, I had overlooked the case `μ {a} = ∞` (I thought you were assuming `μ` to be locally finite, but you're not). In this case, I'm happy with the current formulation. Sorry for the noise!